pyrobosim/pyrobosim/gui/main.py

```diff
@@ -439,6 +448,32 @@ def on_close_click(self) -> None:
         elif (robot is None) and self.goal_textbox.text():
             self.world.close_location(self.goal_textbox.text())
 
+    def on_llm_step_click(self) -> None:
+        """Callback to trigger an LLM-driven policy step."""
+        robot = self.get_current_robot()
+        if robot is None:
+            self.world.logger.warning("No robot selected for LLM control.")
+            return
+        if robot.policy is None:
+            self.world.logger.warning(
+                "Current robot does not have an LLM policy. Launch the demo with --llm-model."
+            )
+            return
+        if robot.is_busy():
+            self.world.logger.info("Robot is busy executing another action.")
+            return
+
+        directive = self.llm_prompt_textbox.text()
+        if hasattr(robot.policy, "set_user_directive"):
+            robot.policy.set_user_directive(directive)
```

Comment:
is there ever a case where the robot policy could _not_ have this attribute? If all policies are from a particular abstract class that implements this method, then not worth doing this check